src/store/checkout-store.js

```diff
@@ -36,12 +38,21 @@ store.registerHandler('CLEAR_ALL_DATA', () => {
 store.registerHandler('NEW_CHECKOUT', data => {
     let checkout = {
         studentId: data.studentId,
-        itemAddresses: data.itemAddresses
+        equipmentAddresses: data.equipmentAddresses
     };
-    
-    data.itemAddresses.forEach(itemAddress => {
-        if (ItemStore.getItemByAddress(itemAddress).status !== 'AVAILABLE') {
-            throw new Error('An item in the cart is not available for checkout.');
+
+    data.equipmentAddresses.forEach(address => {
+        let result = readAddress(address);
+        if(result.type == 'item'){
+            if (ItemStore.getItemByAddress(address).status !== 'AVAILABLE') {
+                throw new Error('An item in the cart is not available for checkout.');
+            }
+        }
+        if(result.type == 'model'){
+            let model = ModelStore.getModelByAddress(address);
+            if(!model.allowCheckout || model.inStock <= 0) {
```

Comment:
This is prevented on the client.  Though it should probably be checked here too, it's difficult because of the fact that the number of a single model checked out isn't provided explicitly.  It's a strong argument for storing checked out models as an object with the model and quantity value, but the client checks should still be sufficient.